// node-red-contrib-mysensors/dist/lib/mysensors-types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var mysensor_command;
(function (mysensor_command) {
    mysensor_command[mysensor_command["C_PRESENTATION"] = 0] = "C_PRESENTATION";
    mysensor_command[mysensor_command["C_SET"] = 1] = "C_SET";
    mysensor_command[mysensor_command["C_REQ"] = 2] = "C_REQ";
    mysensor_command[mysensor_command["C_INTERNAL"] = 3] = "C_INTERNAL";
    mysensor_command[mysensor_command["C_STREAM"] = 4] = "C_STREAM";
})(mysensor_command = exports.mysensor_command || (exports.mysensor_command = {}));
var mysensor_sensor;
(function (mysensor_sensor) {
    mysensor_sensor[mysensor_sensor["S_DOOR"] = 0] = "S_DOOR";
    mysensor_sensor[mysensor_sensor["S_MOTION"] = 1] = "S_MOTION";
    mysensor_sensor[mysensor_sensor["S_SMOKE"] = 2] = "S_SMOKE";
    mysensor_sensor[mysensor_sensor["S_BINARY"] = 3] = "S_BINARY";
    mysensor_sensor[mysensor_sensor["S_LIGHT"] = 3] = "S_LIGHT";
    mysensor_sensor[mysensor_sensor["S_DIMMER"] = 4] = "S_DIMMER";
    mysensor_sensor[mysensor_sensor["S_COVER"] = 5] = "S_COVER";
    mysensor_sensor[mysensor_sensor["S_TEMP"] = 6] = "S_TEMP";
    mysensor_sensor[mysensor_sensor["S_HUM"] = 7] = "S_HUM";
    mysensor_sensor[mysensor_sensor["S_BARO"] = 8] = "S_BARO";
    mysensor_sensor[mysensor_sensor["S_WIND"] = 9] = "S_WIND";
    mysensor_sensor[mysensor_sensor["S_RAIN"] = 10] = "S_RAIN";
    mysensor_sensor[mysensor_sensor["S_UV"] = 11] = "S_UV";
    mysensor_sensor[mysensor_sensor["S_WEIGHT"] = 12] = "S_WEIGHT";
    mysensor_sensor[mysensor_sensor["S_POWER"] = 13] = "S_POWER";
    mysensor_sensor[mysensor_sensor["S_HEATER"] = 14] = "S_HEATER";
    mysensor_sensor[mysensor_sensor["S_DISTANCE"] = 15] = "S_DISTANCE";
    mysensor_sensor[mysensor_sensor["S_LIGHT_LEVEL"] = 16] = "S_LIGHT_LEVEL";
    mysensor_sensor[mysensor_sensor["S_ARDUINO_NODE"] = 17] = "S_ARDUINO_NODE";
    mysensor_sensor[mysensor_sensor["S_ARDUINO_REPEATER_NODE"] = 18] = "S_ARDUINO_REPEATER_NODE";
    mysensor_sensor[mysensor_sensor["S_LOCK"] = 19] = "S_LOCK";
    mysensor_sensor[mysensor_sensor["S_IR"] = 20] = "S_IR";
    mysensor_sensor[mysensor_sensor["S_WATER"] = 21] = "S_WATER";
    mysensor_sensor[mysensor_sensor["S_AIR_QUALITY"] = 22] = "S_AIR_QUALITY";
    mysensor_sensor[mysensor_sensor["S_CUSTOM"] = 23] = "S_CUSTOM";
    mysensor_sensor[mysensor_sensor["S_DUST"] = 24] = "S_DUST";
    mysensor_sensor[mysensor_sensor["S_SCENE_CONTROLLER"] = 25] = "S_SCENE_CONTROLLER";
    mysensor_sensor[mysensor_sensor["S_RGB_LIGHT"] = 26] = "S_RGB_LIGHT";
    mysensor_sensor[mysensor_sensor["S_RGBW_LIGHT"] = 27] = "S_RGBW_LIGHT";
    mysensor_sensor[mysensor_sensor["S_COLOR_SENSOR"] = 28] = "S_COLOR_SENSOR";
    mysensor_sensor[mysensor_sensor["S_HVAC"] = 29] = "S_HVAC";
    mysensor_sensor[mysensor_sensor["S_MULTIMETER"] = 30] = "S_MULTIMETER";
    mysensor_sensor[mysensor_sensor["S_SPRINKLER"] = 31] = "S_SPRINKLER";
    mysensor_sensor[mysensor_sensor["S_WATER_LEAK"] = 32] = "S_WATER_LEAK";
    mysensor_sensor[mysensor_sensor["S_SOUND"] = 33] = "S_SOUND";
    mysensor_sensor[mysensor_sensor["S_VIBRATION"] = 34] = "S_VIBRATION";
    mysensor_sensor[mysensor_sensor["S_MOISTURE"] = 35] = "S_MOISTURE";
    mysensor_sensor[mysensor_sensor["S_INFO"] = 36] = "S_INFO";
    mysensor_sensor[mysensor_sensor["S_GAS"] = 37] = "S_GAS";
    mysensor_sensor[mysensor_sensor["S_GPS"] = 38] = "S_GPS";
    mysensor_sensor[mysensor_sensor["S_WATER_QUALITY"] = 39] = "S_WATER_QUALITY";
})(mysensor_sensor = exports.mysensor_sensor || (exports.mysensor_sensor = {}));
var mysensor_data;
(function (mysensor_data) {
    mysensor_data[mysensor_data["V_TEMP"] = 0] = "V_TEMP";
    mysensor_data[mysensor_data["V_HUM"] = 1] = "V_HUM";
    mysensor_data[mysensor_data["V_STATUS"] = 2] = "V_STATUS";
    mysensor_data[mysensor_data["V_LIGHT"] = 2] = "V_LIGHT";
    mysensor_data[mysensor_data["V_PERCENTAGE"] = 3] = "V_PERCENTAGE";
    mysensor_data[mysensor_data["V_DIMMER"] = 3] = "V_DIMMER";
    mysensor_data[mysensor_data["V_PRESSURE"] = 4] = "V_PRESSURE";
    mysensor_data[mysensor_data["V_FORECAST"] = 5] = "V_FORECAST";
    mysensor_data[mysensor_data["V_RAIN"] = 6] = "V_RAIN";
    mysensor_data[mysensor_data["V_RAINRATE"] = 7] = "V_RAINRATE";
    mysensor_data[mysensor_data["V_WIND"] = 8] = "V_WIND";
    mysensor_data[mysensor_data["V_GUST"] = 9] = "V_GUST";
    mysensor_data[mysensor_data["V_DIRECTION"] = 10] = "V_DIRECTION";
    mysensor_data[mysensor_data["V_UV"] = 11] = "V_UV";
    mysensor_data[mysensor_data["V_WEIGHT"] = 12] = "V_WEIGHT";
    mysensor_data[mysensor_data["V_DISTANCE"] = 13] = "V_DISTANCE";
    mysensor_data[mysensor_data["V_IMPEDANCE"] = 14] = "V_IMPEDANCE";
    mysensor_data[mysensor_data["V_ARMED"] = 15] = "V_ARMED";
    mysensor_data[mysensor_data["V_TRIPPED"] = 16] = "V_TRIPPED";
    mysensor_data[mysensor_data["V_WATT"] = 17] = "V_WATT";
    mysensor_data[mysensor_data["V_KWH"] = 18] = "V_KWH";
    mysensor_data[mysensor_data["V_SCENE_ON"] = 19] = "V_SCENE_ON";
    mysensor_data[mysensor_data["V_SCENE_OFF"] = 20] = "V_SCENE_OFF";
    mysensor_data[mysensor_data["V_HVAC_FLOW_STATE"] = 21] = "V_HVAC_FLOW_STATE";
    mysensor_data[mysensor_data["V_HEATER"] = 21] = "V_HEATER";
    mysensor_data[mysensor_data["V_HVAC_SPEED"] = 22] = "V_HVAC_SPEED";
    mysensor_data[mysensor_data["V_LIGHT_LEVEL"] = 23] = "V_LIGHT_LEVEL";
    mysensor_data[mysensor_data["V_VAR1"] = 24] = "V_VAR1";
    mysensor_data[mysensor_data["V_VAR2"] = 25] = "V_VAR2";
    mysensor_data[mysensor_data["V_VAR3"] = 26] = "V_VAR3";
    mysensor_data[mysensor_data["V_VAR4"] = 27] = "V_VAR4";
    mysensor_data[mysensor_data["V_VAR5"] = 28] = "V_VAR5";
    mysensor_data[mysensor_data["V_UP"] = 29] = "V_UP";
    mysensor_data[mysensor_data["V_DOWN"] = 30] = "V_DOWN";
    mysensor_data[mysensor_data["V_STOP"] = 31] = "V_STOP";
    mysensor_data[mysensor_data["V_IR_SEND"] = 32] = "V_IR_SEND";
    mysensor_data[mysensor_data["V_IR_RECEIVE"] = 33] = "V_IR_RECEIVE";
    mysensor_data[mysensor_data["V_FLOW"] = 34] = "V_FLOW";
    mysensor_data[mysensor_data["V_VOLUME"] = 35] = "V_VOLUME";
    mysensor_data[mysensor_data["V_LOCK_STATUS"] = 36] = "V_LOCK_STATUS";
    mysensor_data[mysensor_data["V_LEVEL"] = 37] = "V_LEVEL";
    mysensor_data[mysensor_data["V_VOLTAGE"] = 38] = "V_VOLTAGE";
    mysensor_data[mysensor_data["V_CURRENT"] = 39] = "V_CURRENT";
    mysensor_data[mysensor_data["V_RGB"] = 40] = "V_RGB";
    mysensor_data[mysensor_data["V_RGBW"] = 41] = "V_RGBW";
    mysensor_data[mysensor_data["V_ID"] = 42] = "V_ID";
    mysensor_data[mysensor_data["V_UNIT_PREFIX"] = 43] = "V_UNIT_PREFIX";
    mysensor_data[mysensor_data["V_HVAC_SETPOINT_COOL"] = 44] = "V_HVAC_SETPOINT_COOL";
    mysensor_data[mysensor_data["V_HVAC_SETPOINT_HEAT"] = 45] = "V_HVAC_SETPOINT_HEAT";
    mysensor_data[mysensor_data["V_HVAC_FLOW_MODE"] = 46] = "V_HVAC_FLOW_MODE";
    mysensor_data[mysensor_data["V_TEXT"] = 47] = "V_TEXT";
    mysensor_data[mysensor_data["V_CUSTOM"] = 48] = "V_CUSTOM";
    mysensor_data[mysensor_data["V_POSITION"] = 49] = "V_POSITION";
    mysensor_data[mysensor_data["V_IR_RECORD"] = 50] = "V_IR_RECORD";
    mysensor_data[mysensor_data["V_PH"] = 51] = "V_PH";
    mysensor_data[mysensor_data["V_ORP"] = 52] = "V_ORP";
    mysensor_data[mysensor_data["V_EC"] = 53] = "V_EC";
    mysensor_data[mysensor_data["V_VAR"] = 54] = "V_VAR";
    mysensor_data[mysensor_data["V_VA"] = 55] = "V_VA";
    mysensor_data[mysensor_data["V_POWER_FACTOR"] = 56] = "V_POWER_FACTOR";
})(mysensor_data = exports.mysensor_data || (exports.mysensor_data = {}));
var mysensor_internal;
(function (mysensor_internal) {
    mysensor_internal[mysensor_internal["I_BATTERY_LEVEL"] = 0] = "I_BATTERY_LEVEL";
    mysensor_internal[mysensor_internal["I_TIME"] = 1] = "I_TIME";
    mysensor_internal[mysensor_internal["I_VERSION"] = 2] = "I_VERSION";
    mysensor_internal[mysensor_internal["I_ID_REQUEST"] = 3] = "I_ID_REQUEST";
    mysensor_internal[mysensor_internal["I_ID_RESPONSE"] = 4] = "I_ID_RESPONSE";
    mysensor_internal[mysensor_internal["I_INCLUSION_MODE"] = 5] = "I_INCLUSION_MODE";
    mysensor_internal[mysensor_internal["I_CONFIG"] = 6] = "I_CONFIG";
    mysensor_internal[mysensor_internal["I_FIND_PARENT_REQUEST"] = 7] = "I_FIND_PARENT_REQUEST";
    mysensor_internal[mysensor_internal["I_FIND_PARENT_RESPONSE"] = 8] = "I_FIND_PARENT_RESPONSE";
    mysensor_internal[mysensor_internal["I_LOG_MESSAGE"] = 9] = "I_LOG_MESSAGE";
    mysensor_internal[mysensor_internal["I_CHILDREN"] = 10] = "I_CHILDREN";
    mysensor_internal[mysensor_internal["I_SKETCH_NAME"] = 11] = "I_SKETCH_NAME";
    mysensor_internal[mysensor_internal["I_SKETCH_VERSION"] = 12] = "I_SKETCH_VERSION";
    mysensor_internal[mysensor_internal["I_REBOOT"] = 13] = "I_REBOOT";
    mysensor_internal[mysensor_internal["I_GATEWAY_READY"] = 14] = "I_GATEWAY_READY";
    mysensor_internal[mysensor_internal["I_SIGNING_PRESENTATION"] = 15] = "I_SIGNING_PRESENTATION";
    mysensor_internal[mysensor_internal["I_NONCE_REQUEST"] = 16] = "I_NONCE_REQUEST";
    mysensor_internal[mysensor_internal["I_NONCE_RESPONSE"] = 17] = "I_NONCE_RESPONSE";
    mysensor_internal[mysensor_internal["I_HEARTBEAT_REQUEST"] = 18] = "I_HEARTBEAT_REQUEST";
    mysensor_internal[mysensor_internal["I_PRESENTATION"] = 19] = "I_PRESENTATION";
    mysensor_internal[mysensor_internal["I_DISCOVER_REQUEST"] = 20] = "I_DISCOVER_REQUEST";
    mysensor_internal[mysensor_internal["I_DISCOVER_RESPONSE"] = 21] = "I_DISCOVER_RESPONSE";
    mysensor_internal[mysensor_internal["I_HEARTBEAT_RESPONSE"] = 22] = "I_HEARTBEAT_RESPONSE";
    mysensor_internal[mysensor_internal["I_LOCKED"] = 23] = "I_LOCKED";
    mysensor_internal[mysensor_internal["I_PING"] = 24] = "I_PING";
    mysensor_internal[mysensor_internal["I_PONG"] = 25] = "I_PONG";
    mysensor_internal[mysensor_internal["I_REGISTRATION_REQUEST"] = 26] = "I_REGISTRATION_REQUEST";
    mysensor_internal[mysensor_internal["I_REGISTRATION_RESPONSE"] = 27] = "I_REGISTRATION_RESPONSE";
    mysensor_internal[mysensor_internal["I_DEBUG"] = 28] = "I_DEBUG";
    mysensor_internal[mysensor_internal["I_SIGNAL_REPORT_REQUEST"] = 29] = "I_SIGNAL_REPORT_REQUEST";
    mysensor_internal[mysensor_internal["I_SIGNAL_REPORT_REVERSE"] = 30] = "I_SIGNAL_REPORT_REVERSE";
    mysensor_internal[mysensor_internal["I_SIGNAL_REPORT_RESPONSE"] = 31] = "I_SIGNAL_REPORT_RESPONSE";
    mysensor_internal[mysensor_internal["I_PRE_SLEEP_NOTIFICATION"] = 32] = "I_PRE_SLEEP_NOTIFICATION";
    mysensor_internal[mysensor_internal["I_POST_SLEEP_NOTIFICATION"] = 33] = "I_POST_SLEEP_NOTIFICATION";
})(mysensor_internal = exports.mysensor_internal || (exports.mysensor_internal = {}));
var mysensor_stream;
(function (mysensor_stream) {
    mysensor_stream[mysensor_stream["ST_FIRMWARE_CONFIG_REQUEST"] = 0] = "ST_FIRMWARE_CONFIG_REQUEST";
    mysensor_stream[mysensor_stream["ST_FIRMWARE_CONFIG_RESPONSE"] = 1] = "ST_FIRMWARE_CONFIG_RESPONSE";
    mysensor_stream[mysensor_stream["ST_FIRMWARE_REQUEST"] = 2] = "ST_FIRMWARE_REQUEST";
    mysensor_stream[mysensor_stream["ST_FIRMWARE_RESPONSE"] = 3] = "ST_FIRMWARE_RESPONSE";
    mysensor_stream[mysensor_stream["ST_SOUND"] = 4] = "ST_SOUND";
    mysensor_stream[mysensor_stream["ST_IMAGE"] = 5] = "ST_IMAGE";
})(mysensor_stream = exports.mysensor_stream || (exports.mysensor_stream = {}));
var mysensor_payload;
(function (mysensor_payload) {
    mysensor_payload[mysensor_payload["P_STRING"] = 0] = "P_STRING";
    mysensor_payload[mysensor_payload["P_BYTE"] = 1] = "P_BYTE";
    mysensor_payload[mysensor_payload["P_INT16"] = 2] = "P_INT16";
    mysensor_payload[mysensor_payload["P_UINT16"] = 3] = "P_UINT16";
    mysensor_payload[mysensor_payload["P_LONG32"] = 4] = "P_LONG32";
    mysensor_payload[mysensor_payload["P_ULONG32"] = 5] = "P_ULONG32";
    mysensor_payload[mysensor_payload["P_CUSTOM"] = 6] = "P_CUSTOM";
    mysensor_payload[mysensor_payload["P_FLOAT32"] = 7] = "P_FLOAT32";
})(mysensor_payload = exports.mysensor_payload || (exports.mysensor_payload = {}));
